test/unit/compute_resources/vmware_test.rb

```diff
@@ -50,25 +50,34 @@ class VmwareTest < ActiveSupport::TestCase
       @cr = FactoryGirl.build(:vmware_cr)
       @cr.stubs(:test_connection)
     end
-    test "calls clone_vm when image provisioning with symbol key" do
-      args = {:image_id =>"2" }
+    test "calls clone_vm when image provisioning with symbol key and provision_method image" do
+      args = {:image_id =>"2", "provision_method" => "image" }
       @cr.stubs(:parse_networks).returns(args)
       @cr.expects(:clone_vm)
       @cr.expects(:new_vm).times(0)
       @cr.create_vm(args)
     end
-    test "calls clone_vm when image provisioning with string key" do
-      args = {"image_id" =>"2" }
+    test "calls clone_vm when image provisioning with string key and provision_method image" do
+      args = {"image_id" =>"2", "provision_method" => "image" }
       @cr.stubs(:parse_networks).returns(args)
       @cr.expects(:clone_vm)
       @cr.expects(:new_vm).times(0)
       @cr.create_vm(args)
     end
+    test "does not call clone_vm when image provisioning with string key and provision_method build" do
+      args = {"image_id" =>"2", "provision_method" => "build" }
+      mock_vm = mock('vm')
+      mock_vm.expects(:save).returns(mock_vm)
+      @cr.stubs(:parse_networks).returns(args)
+      @cr.expects(:clone_vm).times(0)
+      @cr.expects(:new_vm).returns(mock_vm)
+      @cr.create_vm(args)
+    end
   end
 
   test "#create_vm calls clone_vm when image provisioning" do
     attrs_in = HashWithIndifferentAccess.new("image_id"=>"2","cpus"=>"1", "interfaces_attributes"=>{"new_interfaces"=>{"type"=>"VirtualE1000", "network"=>"network-17", "_delete"=>""}, "0"=>{"type"=>"VirtualVmxnet3", "network"=>"network-17", "_delete"=>""}}, "volumes_attributes"=>{"new_volumes"=>{"size_gb"=>"10", "_delete"=>""}, "0"=>{"size_gb"=>"1", "_delete"=>""}})
-    attrs_parsed = HashWithIndifferentAccess.new("image_id"=>"2","cpus"=>"1", "interfaces_attributes"=>{"new_interfaces"=>{"type"=>"VirtualE1000", "network"=>"Test network", "_delete"=>""}, "0"=>{"type"=>"VirtualVmxnet3", "network"=>"Test network", "_delete"=>""}}, "volumes_attributes"=>{"new_volumes"=>{"size_gb"=>"10", "_delete"=>""}, "0"=>{"size_gb"=>"1", "_delete"=>""}})
+    attrs_parsed = HashWithIndifferentAccess.new("image_id"=>"2","cpus"=>"1", "interfaces_attributes"=>{"new_interfaces"=>{"type"=>"VirtualE1000", "network"=>"Test network", "_delete"=>""}, "0"=>{"type"=>"VirtualVmxnet3", "network"=>"Test network", "_delete"=>""}}, "volumes_attributes"=>{"new_volumes"=>{"size_gb"=>"10", "_delete"=>""}, "0"=>{"size_gb"=>"1", "_delete"=>""}}, "provision_method"=>"image")
```

Comment:
The length of the line's gone over a rubocop limit, you'll want to split the hash into multiple lines.